### ISSUE TITLE
Cannot load mazes_tokens from MazeDataset without getting the config

### ISSUE THREAD

**Issue by rusheb** [understanding-search/maze-transformer#53]
This snippet
```python
run_path = "../data/maze/g4-n10"
dataset = MazeDataset.disk_load(run_path, do_tokens=True)
dataset.mazes_tokens
```

Fails with the following error: 
```
'NoneType' object has no attribute 'seq_len_max'
  File "/Users/rusheb/code/maze-transformer/maze_transformer/training/mazedataset.py", line 179, in __init__
    if max_len > cfg.seq_len_max:
  File "/Users/rusheb/code/maze-transformer/maze_transformer/training/mazedataset.py", line 377, in disk_load
    return cls(
  File "/Users/rusheb/Library/Application Support/Code/User/globalStorage/buenon.scratchpads/scratchpads/1a4f2b73c916c9af3926773a674c6453/scratch.py", line 6, in <module>
    dataset = MazeDataset.disk_load(run_path, do_tokens=True)
AttributeError: 'NoneType' object has no attribute 'seq_len_max'
```

The problem is that the `MazeDataset.disk_load` factory method is calling the constructor of `MazeDataset`, which is doing some validation based on the config, which is `None` due to the `do_config` parameter of `disk_load` defaulting to false. 

Naively we could set `do_config` to true, or only do the check if the config is not `None`.  Alternatively 

However, there is quite a lot of complexity in `MazeDataset` that I think we might be able to remove by refactoring. Not sure how valuable this would be.

Points to consider:
* Adding a function to MazeDataset or MazeTokenizer that simply loads the tokens from the tokens file
* Maybe we would want a separate function that just allows us to load the tokens without loading the config
* Using one canonical representation of the MazeDataset and moving the extra work out of the constructor. 
* Simplify the serialization and loading functions by removing the flags (not sure exactly how)
* Where does the check for seq_len_max belong?

**Comment by valedan** [comment]
@mivanit Any updates on ZANJ? Not sure if using that is still the plan or not.

**Comment by mivanit** [comment]
I have actually as of yesterday gotten ZANJ to a working and relatively stable version! Will briefly talk about it in engineering meeting today. I've refactored the whole thing like 4 times but I think this time is the charm.

**Comment by traeuker** [comment]
@mivanit can we close this?

**Comment by mivanit** [comment]
I think this issue can be closed, but the related #163 will require some refactoring of how the tokenization works -- currently too much of it is being handled in `SolvedMaze` and `MazeDataset`.
